hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/StateContext.java

```diff
@@ -415,7 +431,14 @@ public void execute(ExecutorService service, long time, TimeUnit unit)
       if (this.isEntering()) {
         task.onEnter();
       }
-      task.execute(service);
+
+      if (isThreadPoolAvailable(service)) {
+        task.execute(service);
+      } else {
+        LOG.warn("No available thread in pool, duration:" +
```

Comment:
In a similar case of endpoint task locked up, will we end up with flooding of warning logs here?

Reply:
@xiaoyuyao Thanks for review and reasonable suggestion. How about add metrics for thread pool available and unavailable times ?

Reply:
There is a  EndpointStateMachine#logIfNeeded function.   We can leverage this function to reduce the same warning logs logged.  @runzhiwang

Reply:
@ChenSammi @xiaoyuyao  Thanks for review. I have updated the PR, because  EndpointStateMachine#logIfNeeded can not be accessed there, and it's can not be refactored to use, so I did not use it.